packages/genesys-spark-components/src/components/beta/gux-avatar-group/gux-avatar-group.tsx

```diff
@@ -0,0 +1,118 @@
+
+import { trackComponent } from '@utils/tracking/usage';
+import { GuxAvatarAccent } from '../gux-avatar/gux-avatar.types';
+import defaultResources from './i18n/en.json';
+import { GuxAvatarGroupInteractiveElement } from './gux-avatar-groups.types';
+import {
+  first,
+  last,
+  next,
+  previous
+} from 'components/stable/gux-list/gux-list.service';
+
+/**
+ * @slot - slot for a group of gux-avatars, or gux-avatars within button or anchor tags
+ */
+
+const validFocusableItems = ['button', 'a'];
+
+@Component({
+  styleUrl: 'gux-avatar-group.scss',
+  tag: 'gux-avatar-group-beta',
+  shadow: { delegatesFocus: true }
+})
+export class GuxAvatarGroup {
+  private i18n: GetI18nValue;
+
+  @Element()
+  root: HTMLElement;
+
+  @State()
+  count: number = 0;
+
+  @State()
+  avatars: HTMLGuxAvatarBetaElement[];
+
+  @State()
+  interactiveElements: GuxAvatarGroupInteractiveElement[] = [];
+
+  async componentWillLoad(): Promise<void> {
+    this.validateSlot();
+    this.i18n = await buildI18nForComponent(this.root, defaultResources);
+    trackComponent(this.root);
+  }
+
+  private validateSlot(): void {
+    this.interactiveElements = Array.from(
+      this.root.querySelectorAll('a, button')
+    ) as GuxAvatarGroupInteractiveElement[];
+
+    // Disable tabbing on everything but the first interactive element as we're using arrow navigation
+    if (this.interactiveElements.length > 0) {
+      this.interactiveElements.slice(1).forEach(element => {
+        element.tabIndex = -1;
+      });
+    }
+
+    this.avatars = Array.from(
+      this.root.querySelectorAll('gux-avatar-beta')
+    ) as HTMLGuxAvatarBetaElement[];
+
+    if (!this.avatars.length) {
+      logWarn(this.root, 'No avatars found in slot');
+    }
+
+    // This is required for the aria-label
+    this.count = this.avatars.length;
+
+    this.avatars.forEach((avatar: HTMLGuxAvatarBetaElement, index: number) => {
+      // Only small avatars can be included in avatar-group
+      avatar.size = 'small';
+
+      if (avatar.accent === 'default') {
+        avatar.accent = this.getAccent(index) as GuxAvatarAccent;
+      }
+    });
+  }
+
+  @Listen('keydown')
+  onKeyDown(event: KeyboardEvent): void {
+    switch (event.key) {
+      case 'ArrowLeft':
+        event.preventDefault();
+        previous(this.root, validFocusableItems);
+        break;
+      case 'Home':
+        event.preventDefault();
+        first(this.root, validFocusableItems);
+        break;
+      case 'ArrowRight':
+        event.preventDefault();
+        next(this.root, validFocusableItems);
+        break;
+      case 'End':
+        event.preventDefault();
+        last(this.root, validFocusableItems);
+        break;
+    }
+  }
+
+  private getAccent(index): string {
+    const totalAvatarAccentVariations = 12;
+    return `${(index % totalAvatarAccentVariations) + 1}` as GuxAvatarAccent;
+  }
+
+  render(): JSX.Element {
+    return (
+      <div
+        class="gux-avatar-group"
```

Comment:
Should this have a `role` of type `list` also ?

Reply:
I think in this case role `group` with custom directions might be preferable to role `list`. 

There are use cases for an avatar group item that is either just an avatar, an avatar wrapped in a button or an avatar wrapped with a link. 


If I was to use a list item like gux-list-item which needs an interactive element within the `Host`
`
      <Host role="listitem">
        <button type="button" tabIndex={-1} disabled={this.disabled}>
          <slot></slot>
        </button>
      </Host>
`


I can't just slot a button or link in there and have it work the same way. It's important that the original structure like this:
`    <a href="www.google.com" target="_blank">
      <gux-avatar-beta name="Alan Butler">
        <img slot="image" src="https://i.pravatar.cc/300" alt="Alan Butler" />
      </gux-avatar-beta>
    </a>
    `
   is maintained otherwise the link will be read out as a button by a screenreader
   
 Kind of faking it by creating an interactive element inside the list item component to match is not a good way of doing it either. 
 
 So as the least version of evil, I think that using the group and declaring the navigation explicitly might be the most straightforward. Any other thoughts on how to work around this?

Reply:
Group seems fine to me. It doesn't really feel like a list...probably why its called 'avatar-group' & not 'avatar-list'.

Reply:
Looking at it again , I would go with `group` 👍